File: api-cmssignature/api-mssclient/turktelekommssprovider/src/tr/gov/tubitak/uekae/esya/api/webservice/mssclient/turktelekom/status/TurkTelekomStatusResponse.java
package tr.gov.tubitak.uekae.esya.api.webservice.mssclient.turktelekom.status;

import org.etsi.uri.ts102204.v1_1.turktelekom.MSSStatusRespType;
import tr.gov.tubitak.uekae.esya.api.webservice.mssclient.transaction.status.IStatusResponse;
import tr.gov.tubitak.uekae.esya.api.infra.mobile.Status;

/**
 * Turkcell mobile signature status response implementation
 * @see tr.gov.tubitak.uekae.esya.api.webservice.mssclient.transaction.status.IStatusResponse
 */
public class TurkTelekomStatusResponse implements IStatusResponse {
    MSSStatusRespType mssStatusResponse;

    public TurkTelekomStatusResponse(MSSStatusRespType mssStatusResponse) {
        this.mssStatusResponse = mssStatusResponse;
    }

    public String getMSISDN() {
        return mssStatusResponse.getMobileUser().getMSISDN();
    }

    public Status getStatus() {
        return new Status(mssStatusResponse.getStatus().getStatusCode().getValue().toString(), mssStatusResponse.getStatus().getStatusMessage());
    }

    public byte[] getSignature() {
        return mssStatusResponse.getMSSSignature().getBase64Signature();
    }
}

File: api-cmssignature/api-mssclient/turktelekommssprovider/csharp/src/turktelekom/status/TurkTelekomStatusResponse.cs
using System;
using tr.gov.tubitak.uekae.esya.api.webservice.mssclient.turktelekom.stub.status;
using tr.gov.tubitak.uekae.esya.api.webservice.mssclient.transaction.status;
using tr.gov.tubitak.uekae.esya.api.webservice.mssclient.wrapper;

namespace tr.gov.tubitak.uekae.esya.api.webservice.mssclient.turktelekom.status
{
    /**
 * Turkcell mobile signature status response implementation
 * @see IStatusResponse
 */

    public class TurkTelekomStatusResponse : IStatusResponse
    {
        private readonly MSS_StatusRespType _statusResp;

        public TurkTelekomStatusResponse(MSS_StatusRespType aStatusResponse)
        {
            _statusResp = aStatusResponse;
        }

        public String getMSISDN()
        {
            return _statusResp.MobileUser.MSISDN;
        }

        public Status getStatus()
        {
            return new Status(_statusResp.Status.StatusCode.Value, _statusResp.Status.StatusMessage);
        }

        public byte[] getSignature()
        {
            return (byte[]) _statusResp.MSS_Signature.Base64Signature;
        }
    }
}